src/libraries/System.Private.CoreLib/src/System/Byte.cs

```diff
@@ -272,5 +279,704 @@ object IConvertible.ToType(Type type, IFormatProvider? provider)
+                {
+                    result = default;
+                    return false;
+                }
+
+                result = (byte)actualValue;
+                return true;
+            }
+            else if (typeof(TOther) == typeof(ushort))
+            {
+                var actualValue = (ushort)(object)value;
+
+                if (actualValue > MaxValue)
+                {
+                    result = default;
+                    return false;
+                }
+
+                result = (byte)actualValue;
+                return true;
+            }
+            else if (typeof(TOther) == typeof(uint))
+            {
+                var actualValue = (uint)(object)value;
+
+                if (actualValue > MaxValue)
+                {
+                    result = default;
+                    return false;
+                }
+
+                result = (byte)actualValue;
+                return true;
+            }
+            else if (typeof(TOther) == typeof(ulong))
+            {
+                var actualValue = (ulong)(object)value;
+
+                if (actualValue > MaxValue)
+                {
+                    result = default;
+                    return false;
+                }
+
+                result = (byte)actualValue;
+                return true;
+            }
+            else if (typeof(TOther) == typeof(nuint))
+            {
+                var actualValue = (nuint)(object)value;
+
+                if (actualValue > MaxValue)
+                {
+                    result = default;
+                    return false;
+                }
+
+                result = (byte)actualValue;
+                return true;
+            }
+            else
+            {
+                ThrowHelper.ThrowNotSupportedException();
+                result = default;
+                return false;
+            }
+        }
+
+        //
+        // IParseable
+        //
+
+        [RequiresPreviewFeatures]
+        static bool IParseable<byte>.TryParse([NotNullWhen(true)] string? s, IFormatProvider? provider, out byte result)
+            => TryParse(s, NumberStyles.Integer, provider, out result);
+
+        //
+        // IShiftOperators
+        //
+
+        [RequiresPreviewFeatures]
+        static byte IShiftOperators<byte, byte>.operator <<(byte value, int shiftAmount)
+            => (byte)(value << shiftAmount);
+
+        [RequiresPreviewFeatures]
+        static byte IShiftOperators<byte, byte>.operator >>(byte value, int shiftAmount)
+            => (byte)(value >> shiftAmount);
+
+        // [RequiresPreviewFeatures]
+        // static byte IShiftOperators<byte, byte, byte>.operator >>>(byte value, int shiftAmount)
+        //     => (byte)(value >> shiftAmount);
+
+        //
+        // ISpanParseable
+        //
+
+        [RequiresPreviewFeatures]
+        static byte ISpanParseable<byte>.Parse(ReadOnlySpan<char> s, IFormatProvider? provider)
```

Comment:
I thought we'd discussed just combining ISpanParseable and IParseable.  Did we decide not to?

Reply:
We hadn't formalized this in API review and there were still open questions here about F# and VB to my understanding.

Reply:
When are we going to decide on it then?

Reply:
Sometime before this ships as stable, so possibly after .NET 6.